pyvex/expr.py

```diff
@@ -15,14 +20,14 @@ class IRExpr(VEXObject):
 
     __slots__ = [ ]
 
-    tag = None
+    tag = None # type: Optional[str]
     tag_int = 0  # set automatically at bottom of file
 
     def pp(self):
         print(self.__str__())
 
     @property
-    def child_expressions(self):
+    def child_expressions(self) -> List['IRExpr']:
```

Comment:
I'm not sure that this is valid, shouldn't this be some IRExpr type rather than the literal string "IRExpr" ?

Reply:
It is valid.

Reply:
This is a [forward reference](https://www.python.org/dev/peps/pep-0484/#forward-references). This is required in various places because `def child_expressions(self) -> List[IRExpr]` would require the type to be available when this definition is parsed. The import only happens later (or not at all?), so this would be a syntax error. This is needed in various places and still properly type checks

Reply:
Ahh, gotcha. TIL, thanks for the explanation!